### DISCUSSION TITLE
Ability to create form on posgres custom schema - DONE

### DISCUSSION THREAD

**Discussion by SG2019** [Budibase/budibase#2586] [Ideas]
Hi, Is it possible to create form and save rows on tables that are in posgres custom schema  ? On adding action button , the table does not display under Table fropdown in Save Row action. I can see only public schema's table or budibase internal db tables.

**Comment by mjashanks** [comment]
Hi @SG2019, this is not possible at the moment. How would expect this to work? Should the `schema` be one of the properties that is set up with the initial connection details? Or should all tables from all schemas appear under the one connection?

Would be good to hear your opinions on how it should be done, and why.

Thanks!

**Reply by SG2019** [reply]
Thankyou @mjashanks for the update. 

I am thinking more on option2 where all tables are available from all the schemas under the databse. 

This will help in creating and editing the form and no need to create separate connections for non public schemas.

**Reply by mjashanks** [reply]
Thanks, have converted this to an Idea so that it can get considered for future development.

I'd suggest that we
1. Have an optional `Schema` input, on the connection details screen
2. If the schema is entered, we include only tables from that schema
3. If no schema is entered, we list tables from all schemas

**Reply by richardalmonte** [reply]
It would be nice having the same feature for SQL Server too. Today (2022/02/16) it is not possible. 

Thanks

**Reply by mshiner** [reply]
Is there any movement on using tables from other than the public schema? (@mjshanks reply 13/09/2021).

We have a postgreSQL ERP system where generic system tables are in public (the default) with reporing and site specific tables in their own schema.

To build on @mjashanks answer - a 'tick box' to select 1 or more specific schema or 'all' would be great.

As an aside we'ev connected up LibreOffice Base before and that even allows for a 'check box ' type arrangement on invidual tables as well - which is neat.

Regards
M
